lib/ansible/plugins/lookup/manifold.py

```diff
@@ -0,0 +1,276 @@
+            type: string
+            required: False
+        team:
+            description:
+                - The team label you want to get the resource for.
+            type: string
+            required: False
+'''
+
+EXAMPLES = '''
+    - name: all available resources
+      debug: msg="{{ lookup('manifold', api_token='SecretToken') }}"
+    - name: all available resources for a specific project in specific team
+      debug: msg="{{ lookup('manifold', api_token='SecretToken', project='poject-1', team='team-2') }}"
+    - name: two specific resources
+      debug: msg="{{ lookup('manifold', 'resource-1', 'resource-2') }}"
+'''
+
+RETURN = '''
+    _raw:
+        description:
+            - dictionary of credentials ready to be consumed as environment variables. If multiple resources define
+              the same environment variable(s), the last one returned by the Manifold API will take precedence.
+        type: dict
+'''
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.urls import open_url, ConnectionError, SSLValidationError
+from ansible.module_utils.six.moves.urllib.error import HTTPError, URLError
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible.module_utils import six
+from ansible.utils.display import Display
+from traceback import format_exception
+import json
+import sys
+import os
+
+display = Display()
+
+
+class ApiError(Exception):
+    pass
+
+
+class ManifoldApiClient(object):
+    base_url = 'https://api.{api}.manifold.co/v1/{endpoint}'
+    http_agent = 'python-manifold-ansible-1.0.0'
+
+    def __init__(self, token):
+        self._token = token
+
+    def request(self, api, endpoint, *args, **kwargs):
+        """
+        Send a request to API backend and pre-process a response.
+        :param api: API to send a request to
+        :type api: str
+        :param endpoint: API endpoint to fetch data from
+        :type endpoint: str
+        :param args: other args for open_url
+        :param kwargs: other kwargs for open_url
+        :return: server response. JSON response is automatically deserialized.
+        :rtype: dict | list | str
+        """
+
+        default_headers = {
+            'Authorization': "Bearer {0}".format(self._token),
+            'Accept': "*/*"  # Otherwise server doesn't set content-type header
+        }
+
+        url = self.base_url.format(api=api, endpoint=endpoint)
+
+        headers = default_headers
+        arg_headers = kwargs.pop('headers', None)
+        if arg_headers:
+            headers.update(arg_headers)
+
+        try:
+            display.vvvv('manifold lookup connecting to {0}'.format(url))
+            response = open_url(url, headers=headers, http_agent=self.http_agent, *args, **kwargs)
+            data = response.read()
+            if response.headers.getheader('content-type') == 'application/json':
```

Comment:
This line fails for me trying to run Ansible under Python3 - `response.headers` is there, but there is no `getheader` method.  I don't know the internals of what is returned from `open_url` but I can call `get` on it and it works for both Python2 and Python3:

```
fatal: [127.0.0.1]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'manifold'. Error was a <class 'ansible.errors.AnsibleError'>, original message: ['Traceback (most recent call last):\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 226, in run\\n    team_data = client.get_teams(team)\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 166, in get_teams\\n    data = self.request(api, endpoint)\\n', '  File \"/Users/dave/work/src/ansible-test/ansible/lib/ansible/plugins/lookup/manifold.py\", line 113, in request\\n    if response.headers.getheader(\\'content-type\\') == \\'application/json\\':\\n', \"AttributeError: 'HTTPMessage' object has no attribute 'getheader'\\n\"]"
}
```

Reply:
@davidharrigan Nice spot, thanks.
From a quick Google I've found
* https://github.com/TAXIIProject/libtaxii/commit/59e18912e90550e2248779518fb63fac77d2f5a1
* https://bugs.python.org/issue4773

Reply:
Thank you both! @galanoff plans to get those fixes in. @gundalow do we have a deadline for 2.8?